README.md

```diff
@@ -1,65 +1,227 @@
+
+A lot of existing sidebars have a ton of ERB tags that call a `sidebar_current` method. Ignore these, and don't worry about including them when making updates. They were for a hack that isn't needed anymore.
+
+You don't need to add anything special to a sidebar to get the dynamic JavaScript open/close behavior, but note that the "expand all" and filter controls are only added for sidebars with more than a certain number of links.
+
+## Living With Submodules
+
+[inpage-submodules]: #living-with-submodules
+
+Using submodules lets us keep the docs for providers and Terraform core right next to the code, which is very helpful for the engineers who do most of the docs updates. However, it imposes some extra costs when working with this repo. Here's how to get around the worst of those costs.
+
+### Basics
+
+In your local checkout of this repo, Git submodules can be active or sleeping. The `git submodule init <PATH>` and `git submodule deinit <PATH>` commands switch them between the two states.
+
+Once you `init` a submodule, you usually need to run `git submodule update`, which will either do the initial checkout or update the working copy to the commit that `terraform-website` currently expects.
+
+If a submodule shows up as "changed" in `git status` but you haven't done anything with it, it probably just means you updated your `terraform-website` branch and it now says the submodule should be at a newer commit. Run `git submodule update` to resolve it. Inactive submodules don't show up in `git status`.
+
+In general, you should never need to commit a submodule update to `terraform-website`; updates get handled automatically during provider releases, and aren't necessary for getting new content onto the website anyway.
+
+Avoid running `git rm` on a submodule unless you know what you're doing. You usually want `git submodule deinit` instead.
+
+### Don't Keep Every Submodule Active
+
+Earlier instructions for working with this repo said to use `git submodule init` (with no `<PATH>` argument) or `git submodule update --init` or `make sync` to activate everything. **Don't do that.** Git commands will take forever to run, and if your `$PS1` includes hints about the current directory's Git status, your entire terminal will slow to a **c r a w l.**
+
+Instead, only init the specific submodules you currently need to work with (`git submodule init ext/providers/aws`), and feel free to de-init them when you're done. De-initting is non-destructive as long as you've committed your changes within the submodule (and preferably pushed your branch) -- Git keeps the repository data cached out of the way, so it dosen't even need to clone the entire repo again the next time you init it.
+
+If you previously activated a hundred submodules and regret it, you can run `git submodule deinit --all` or `make deinit` to start fresh.
```

Comment:
Oh my goodness this section is _life changing_ 🙌

Reply:
I've been meaning to write this stuff down for ages, IDEK why it took me this long, but I'm glad it's out there now!